xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/attachment_macros.vm

```diff
@@ -247,37 +246,37 @@
   </div>
 #end
 
-## Display a livetable with attachments of the specified document. Use the given id when displaying the livetable and
-## for the javascript function name.
-#macro (showAttachmentsLivetable $attachmentsDoc $livetableId)
-  #set ($javascriptName = $livetableId + 'Livetable')
-  #set ($columns = ['mimeType', 'filename', 'filesize', 'date', 'author', 'actions'])
-  #set ($columnsProperties = {
-    'filename': {'type': 'text', 'html': true},
-    'date': {'type': 'date'},
-    'author': {'type': 'text', 'html': true},
-    'actions': {'sortable': false,'filterable': false,'html': true}
+## Display a liveData with attachments from the specified document.
+#macro (showAttachmentsLiveData $attachmentsDoc $liveDataId)
+  #set ($liveDataConfig = {
+    'meta': {
+      'propertyDescriptors': [
+        { 'id': 'mimeType', 'displayer': 'html'},
+        { 'id': 'filename', 'displayer': 'html' },
+        { 'id': 'filesize', 'displayer': 'html' },
+        { 'id': 'date', 'filter': 'date'},
+        { 'id': 'author', 'displayer': 'html' },
+        { 'id': 'actions', 'displayer': 'html' }
+      ],
+      'entryDescriptor': {
+        'idProperty': 'id'
+      }
+    }
   })
-  #if ($xwiki.exists('XWiki.AllAttachments'))
-    #set ($discard = $columnsProperties.putAll({
-      'mimeType': {'type': 'list', 'html': true, 'class': 'XWiki.AllAttachments'},
-      'filesize': {'type': 'list', 'html': true, 'class': 'XWiki.AllAttachments'}
-    }))
-  #else
-    ## When the class cannot be used, fallback to other filtering options, as appropriate by case.
-    #set ($discard = $columnsProperties.putAll({
-      'mimeType': {'type': 'text', 'html': true},
-      'filesize': {'html': true, 'filterable': false}
-    }))
-  #end
-  #set ($options = {
+  #set ($sourceParameters = $escapetool.url({
+    'template': 'xpart.vm',
     'translationPrefix': 'core.viewers.attachments.livetable.',
-    'url': $attachmentsDoc.getURL('get', "xpage=xpart&vm=attachmentsjson.vm"),
-    'rowCount': 5,
-    'selectedColumn': 'filename',
-    'javascriptName': $javascriptName,
-    'outputOnlyHtml': true
-  })
-  #livetable($livetableId $columns $columnsProperties $options)
-  #deleteAttachmentModal
+    'className': 'XWiki.AllAttachments',
+    "\$doc": "$attachmentsDoc",
```

Comment:
I think `'$doc'` would work as well, without having to escape the `$`.

Reply:
I get an `updating entries failed` error from the liveData if I don't use escaping